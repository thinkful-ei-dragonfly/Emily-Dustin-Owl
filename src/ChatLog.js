import React from 'react';
import ChatEvent from './ChatEvent';



export default function(props) {
  return (
    <div className="participant_list">
      {props.chatEvents.map((event,i) => (
      <ChatEvent 
      key = {i} 
      participantId={event.participantId} 
      type={event.type}
      time={event.time}
      message={event.message}/>))}
    </div>
  );
}
