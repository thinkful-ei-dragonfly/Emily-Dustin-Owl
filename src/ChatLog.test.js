import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatLog from './ChatLog';



describe('Chat Log component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    // eslint-disable-next-line react/jsx-pascal-case
    ReactDOM.render(<ChatLog chatEvents={[
      {
          participantId: 3,
          type: 'message',
          message: 'Hello world',
          time: 1548852646559,
          timestamp: 1548852484247
      },
      {
          participantId: 2,
          type: 'thumbs-up',
          timestamp: 1548852484247
      },
      {
          participantId: 4,
          type: 'thumbs-down',
          timestamp: 1548852484247
      }]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expexted', () => {
    const tree = renderer
      // eslint-disable-next-line react/jsx-pascal-case
      .create(<ChatLog chatEvents={[
        {
            participantId: 3,
            type: 'message',
            message: 'Hello world',
            time: 1548852646559,
            timestamp: 1548852484247
        },
        {
            participantId: 2,
            type: 'thumbs-up',
            timestamp: 1548852484247
        },
        {
            participantId: 4,
            type: 'thumbs-down',
            timestamp: 1548852484247
        }]}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});