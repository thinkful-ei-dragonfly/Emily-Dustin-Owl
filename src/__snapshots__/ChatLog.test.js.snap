// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Log component renders the UI as expexted 1`] = `
<div
  className="participant_list"
>
  <div
    className="message"
  >
    <img
      alt="user avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <h3>
      Axe Kubicka
    </h3>
    <p>
      5:50:46 AM
    </p>
    <p
      className="message-text"
    >
      Hello world
    </p>
  </div>
  <div
    className="thumbs-up"
  >
    <p>
      <strong>
        Caty Flucker
      </strong>
      gave a thumbs up
      <i
        className="fas fa-thumbs-up"
      />
    </p>
  </div>
  <div
    className="thumbs-down"
  >
    <p>
      <strong>
        Frank Runciman
      </strong>
      gave a thumbs down
      <i
        className="fas fa-thumbs-down"
      />
    </p>
  </div>
</div>
`;
